app/models/source_type.rb

```diff
@@ -1,3 +1,4 @@
 class SourceType < ApplicationRecord
   has_many :sources
+  has_many :availabilities, :as => :resource, :dependent => :destroy
```

Comment:
@agrare I wonder what the example with source type will be? Using some kind of inheritance? Like:

`if source.source_type.available?(feature) && source.available(feature)?`

Or would we always copy it to source when adding it? The source_type will probably have only `supported/unsupported`

Reply:
I was thinking we would copy the source_type availabilities when the source was created

Reply:
hm, copying might not be the best as we add more stuff, but lets solve that when we get to it. It won't be hard to migrate to something else